src/state/zustand/swapperStore/amountSelectors.ts

```diff
@@ -262,10 +264,27 @@ export const selectSellAmountPlusFeesFiat = createSelector(
   },
 )
 
-export const selectQuoteSellAmountBeforeFeesSellAssetBaseUnit = createSelector(
+export const selectTradeOrQuoteSellAmountBeforeFeesCryptoBaseUnit = createSelector(
   (state: SwapperState) =>
     state.activeSwapperWithMetadata?.quote?.sellAmountBeforeFeesCryptoBaseUnit,
-  (quoteSellAmountBeforeFeesBaseUnit): string | undefined => quoteSellAmountBeforeFeesBaseUnit,
+  (state: SwapperState) => state.trade?.sellAmountBeforeFeesCryptoBaseUnit,
+  (quoteSellAmountBeforeFeesBaseUnit, tradeSellAmountBeforeFeesBaseUnit): string | undefined =>
+    // Use the trade amount if we have it, otherwise use the quote amount
+    tradeSellAmountBeforeFeesBaseUnit ?? quoteSellAmountBeforeFeesBaseUnit,
+)
+
+export const selectTradeOrQuoteBuyAmountCryptoBaseUnit = createSelector(
+  (state: SwapperState) => state.activeSwapperWithMetadata?.quote?.buyAmountCryptoBaseUnit,
+  (state: SwapperState) => state.trade?.buyAmountCryptoBaseUnit,
+  (quoteBuyAmountBeforeFeesBaseUnit, tradeBuyAmountBeforeFeesBaseUnit): string | undefined =>
+    // Use the trade amount if we have it, otherwise use the quote amount
+    tradeBuyAmountBeforeFeesBaseUnit ?? quoteBuyAmountBeforeFeesBaseUnit,
+)
+
+export const selectQuoteSellAmountBeforeFeesSellAssetBaseUnit = createSelector(
+  selectTradeOrQuoteSellAmountBeforeFeesCryptoBaseUnit,
+  (tradeOrQuoteSellAmountBeforeFeesCryptoBaseUnit): string | undefined =>
+    tradeOrQuoteSellAmountBeforeFeesCryptoBaseUnit,
 )
```

Comment:
if this is just for clarity, can we avoid creating another selector by just assigning it?

```suggestion
export const selectQuoteSellAmountBeforeFeesSellAssetBaseUnit 
  selectTradeOrQuoteSellAmountBeforeFeesCryptoBaseUnit
```

Reply:
Looks like I missed this after a refactor, it doesn't need to exist at all!

Removed in https://github.com/shapeshift/web/pull/4353/commits/47b3c94b52c5a0bb478160ced5b91a2c9b764122.